rswag-api/lib/rswag/api/middleware.rb

```diff
@@ -13,7 +13,11 @@ def initialize(app, config)
 
       def call(env)
         path = env['PATH_INFO']
-        filename = "#{@config.resolve_swagger_root(env)}/#{path}"
+        # Sanitize the filename for directory traversal by expanding, and matching the swagger root directory
+        filename = File.expand_path(File.join(@config.resolve_swagger_root(env), path))
+        unless filename.match Regexp.new('^' + Regexp.escape(@config.resolve_swagger_root(env)))
```

Comment:
Just a thought, could this just use 'start_with?', or if using regex should the expression maybe use '\A' rather than '^'?

Reply:
`start_with?` is much more ruby like, I like it. While checking it also noticed that[ `expand_path` also does the file ](https://www.rubydoc.info/stdlib/core/File.expand_path) joining, so I can remove that File.join with the `dir_string` param.

I'll merge if tests pass after the changes.

Reply:
ok thanks!

Reply:
@sergioisidoro Unfortunately, the `dir_string` will be ignored in case the file name starts with a slash (as is the case with absolute paths, but also in our case of getting the path out of `env['PATH_INFO']`). So the "simplify file expansion" part broke the functionality of the gem. Will post a PR to fix this.

Reply:
#660